packages/request-state/src/index.ts

```diff
@@ -0,0 +1,24 @@
+import { defineIntegration, addVitePlugin } from 'astro-integration-kit';
+import { plugin } from './plugin.js';
+
+export default defineIntegration({
+	name: '@inox-tools/request-state',
+	setup() {
+		return {
+			hooks: {
+				'astro:config:setup': (params) => {
+					const { addMiddleware } = params;
+
+					addMiddleware({
+						order: 'pre',
+						entrypoint: '@inox-tools/request-state/runtime/middleware',
+					});
+					addVitePlugin(params, {
+						warnDuplicated: true,
+						plugin: plugin(),
```

Comment:
is there a reason for the plugin to be a function?

Reply:
It has two benefits:
- If it needs arguments in the future it is less changes to make
- It only instantiates if actually used

Both are quite negligible on such a tiny plugin, but I'm more used to it as a function.